### PR TITLE
fix: update me and users routes to match cloud/documentation

### LINKED ISSUE #23716: `PUT /api/v2/me/password` doesn't update the user password
A request to `PUT /api/v2/me/password` with the session cookie, basic auth credentials, and password in the body results in a `500` status and _two_ error objects in complaining that `user ID` is missing.

The influxdb [`handlePutUserPassword` handler](https://github.com/influxdata/influxdb/blob/b0a0e734e0c46384499cdc1f08f2673b6a28ff98/tenant/http_server_user.go#L126) appears to throw an error whenever the URL doesn't contain a user ID, even though user_id is apparently in the context, if I'm interpreting it correctly.

[Comment from @jeffreyssmith2nd](https://influxdata.slack.com/archives/CPYJNHM4K/p1662575105805159?thread_ts=1662573605.214989&cid=CPYJNHM4K):
Just from a quick look, it seems like /api/v2/me/password and /api/v2/:id/password are using the same handler and that is why you're seeing that behaviour. I would assume this is a bug, and that me/password shouldn't require an id

__Steps to reproduce:__
List the minimal actions needed to reproduce the behavior.

1. Pass the user's _Basic authentication_ credentials (`USERNAME:PASSWORD`) to the `POST /api/v2/signin`
    endpoint to create a user session and generate a session cookie.
2. From the response in the first step, extract the session cookie (`Set-Cookie`) header.
3. Pass the following in a request to the `PUT /api/v2/me/password` endpoint:
     - The `Set-Cookie` header from the second step
     - The `Authorization Basic` header with the user's _Basic authentication_ credentials
     - `{"password": "NEW_PASSWORD"}` in the request body

__Expected behavior:__

There's no Grace test for this method.

According to the openapi spec, `PUT /api/v2/me/password` requires basic auth (username and password in the `Authorization` header`) and {"password": "NEW_PASSWORD"} in the request body. It also requires a user session cookie, although that isn't documented. It should not require a user ID in the request.

It should respond with a `204` status and the user password should be updated to the `NEW_PASSWORD`.

__Actual behavior:__

`500` status with the following response:

```
< HTTP/1.1 500 Internal Server Error
< Content-Type: application/json; charset=utf-8
< X-Influxdb-Build: OSS
< X-Influxdb-Version: 2.3.0
< X-Platform-Error-Code: internal error
< Date: Tue, 06 Sep 2022 21:30:45 GMT
< Content-Length: 149
< 
{ [149 bytes data]
100   180  100   149  100    31   5321   1107 --:--:-- --:--:-- --:--:--  6428
* Connection #0 to host localhost left intact
{
  "code": "internal error",
  "message": "invalid user ID provided in route"
}
{
  "code": "invalid",
  "message": "id must have a length of 16 bytes"
}
```

*cURL request*:

```sh
* Connected to localhost (::1) port 8086 (#0)
* Server auth using Basic with user 'jstirnaman'
> PUT /api/v2/me/password HTTP/1.1
> Host: localhost:8086
> Authorization: Basic anN0aXJuYW1hbjpwYXNzd29yZA==
> User-Agent: curl/7.71.1
> Accept: */*
> Cookie: influxdb-oss-session=19vduZYGOvP2GGryXVT2qYftlFKu3bIopurM6AGFow1yF1abhtOlbHfsc-d8gozZFC_6WxmlQIAwLMW5xs523w==
> Content-Length: 31
> Content-Type: application/x-www-form-urlencoded
> 
} [31 bytes data]
* upload completely sent off: 31 out of 31 bytes
* Mark bundle as not supporting multiuse
< HTTP/1.1 500 Internal Server Error
< Content-Type: application/json; charset=utf-8
< X-Influxdb-Build: OSS
< X-Influxdb-Version: 2.3.0
< X-Platform-Error-Code: internal error
< Date: Tue, 06 Sep 2022 21:30:45 GMT
< Content-Length: 149
< 
{ [149 bytes data]
100   180  100   149  100    31   5321   1107 --:--:-- --:--:-- --:--:--  6428
* Connection #0 to host localhost left intact
{
  "code": "internal error",
  "message": "invalid user ID provided in route"
}
{
  "code": "invalid",
  "message": "id must have a length of 16 bytes"
}
```
__Environment info:__

* System info: Run `uname -srm` and copy the output here
* InfluxDB version: Run `influxd version` and copy the output here

  ```sh                                            
  InfluxDB dev (git: none) build_date: 2022-09-07T20:29:40Z
  ```
 
Also tested on v2.3.0 with the same result.

* Other relevant environment details: Container runtime, disk info, etc

__Config:__
Copy any non-default config values here or attach the full config as a gist or file.

__Logs:__

*Influxd log:*

```sh
2022-09-07T17:32:19.296957Z     info    request {"log_id": "0cnFyGv0000", "path": "/api/v2/signin", "filename": "net/http/request.go", "line": 356, "error.object": "no span ID found in carrier", "user_agent": "curl", "Accept": "*/*", "newLabels": false, "memoryOptimizedFill": false, "queryTracing": false, "cursorAtEOF": false, "groupTransformationGroup": false, "narrowTransformationFill": false, "vectorizeAddition": false, "optimizeSetTransformation": false, "refreshSingleCell": true, "optimizeAggregateWindow": false, "narrowTransformationLimit": false, "appMetrics": false, "optimizeStateTracking": false, "enforceOrgDashboardLimits": false, "queryConcurrencyLimit": 0, "aggregateTransformationTransport": false, "vectorizeLogicalOperators": false, "unusedSymbolWarnings": false, "groupWindowAggregateTranspose": false, "timeFilterFlags": false, "optimizeUnionTransformation": false, "vectorizedMap": false, "vectorizeOperators": false, "labelPolymorphism": false, "memoryOptimizedSchemaMutation": false, "injectLatestSuccessTime": false, "newAutoRefresh": true, "narrowTransformationDifference": false, "ot_trace_id": "09f23ffa65623002", "ot_span_id": "09f23ffa65623001", "ot_start": "2022-09-07T17:32:19.221273Z", "ot_stop": "2022-09-07T17:32:19.296955Z", "ot_duration": "75.708ms", "method": "POST", "route": "", "handler": "platform"}
2022-09-07T17:32:19.315972Z     info    bolt.(*KVStore).View    {"log_id": "0cnFyGv0000", "filename": "github.com/influxdata/influxdb/v2/bolt/kv.go", "line": 164, "ot_trace_id": "09f23ffa7ce23002", "ot_span_id": "09f23ffa7ce23003", "ot_start": "2022-09-07T17:32:19.315649Z", "ot_stop": "2022-09-07T17:32:19.315967Z", "ot_duration": "0.318ms", "ot_child_of": "09f23ffa7ce23001"}
2022-09-07T17:32:19.338164Z     info    bolt.(*KVStore).View    {"log_id": "0cnFyGv0000", "filename": "github.com/influxdata/influxdb/v2/bolt/kv.go", "line": 164, "ot_trace_id": "09f23ffa7ce23002", "ot_span_id": "09f23ffa7d223000", "ot_start": "2022-09-07T17:32:19.316848Z", "ot_stop": "2022-09-07T17:32:19.338152Z", "ot_duration": "21.311ms", "ot_child_of": "09f23ffa7ce23001"}
2022-09-07T17:32:19.338597Z     info    bolt.(*KVStore).View    {"log_id": "0cnFyGv0000", "filename": "github.com/influxdata/influxdb/v2/bolt/kv.go", "line": 164, "ot_trace_id": "09f23ffa7ce23002", "ot_span_id": "09f23ffa82a23000", "ot_start": "2022-09-07T17:32:19.338577Z", "ot_stop": "2022-09-07T17:32:19.338595Z", "ot_duration": "0.018ms", "ot_child_of": "09f23ffa7ce23001"}
2022-09-07T17:32:19.338955Z     info    bolt.(*KVStore).View    {"log_id": "0cnFyGv0000", "filename": "github.com/influxdata/influxdb/v2/bolt/kv.go", "line": 164, "ot_trace_id": "09f23ffa7ce23002", "ot_span_id": "09f23ffa82a23001", "ot_start": "2022-09-07T17:32:19.338876Z", "ot_stop": "2022-09-07T17:32:19.338952Z", "ot_duration": "0.076ms", "ot_child_of": "09f23ffa7ce23001"}
2022-09-07T17:32:19.339156Z     info    request {"log_id": "0cnFyGv0000", "path": "/api/v2/me", "filename": "net/http/request.go", "line": 356, "error.object": "no span ID found in carrier", "user_agent": "curl", "Accept": "*/*", "Cookie": "influxdb-oss-session=GU6DzwdrK7GFo8D0CAxECQlSxkTViMlO1hscEn67S7vZ1dGcHkjv84vrllzLlq7YGHtzAWRSqoGhfwe9l0pwsg==", "vectorizedMap": false, "labelPolymorphism": false, "unusedSymbolWarnings": false, "appMetrics": false, "memoryOptimizedSchemaMutation": false, "newAutoRefresh": true, "aggregateTransformationTransport": false, "groupTransformationGroup": false, "queryConcurrencyLimit": 0, "injectLatestSuccessTime": false, "optimizeAggregateWindow": false, "cursorAtEOF": false, "refreshSingleCell": true, "narrowTransformationFill": false, "memoryOptimizedFill": false, "queryTracing": false, "narrowTransformationLimit": false, "newLabels": false, "enforceOrgDashboardLimits": false, "optimizeUnionTransformation": false, "narrowTransformationDifference": false, "vectorizeAddition": false, "vectorizeOperators": false, "optimizeSetTransformation": false, "groupWindowAggregateTranspose": false, "timeFilterFlags": false, "optimizeStateTracking": false, "vectorizeLogicalOperators": false, "ot_trace_id": "09f23ffa7ce23002", "ot_span_id": "09f23ffa7ce23001", "ot_start": "2022-09-07T17:32:19.315545Z", "ot_stop": "2022-09-07T17:32:19.339154Z", "ot_duration": "23.617ms", "method": "GET", "route": "", "handler": "platform", "user_id": "0772396d1f411000"}
2022-09-07T17:32:19.355492Z     info    bolt.(*KVStore).View    {"log_id": "0cnFyGv0000", "filename": "github.com/influxdata/influxdb/v2/bolt/kv.go", "line": 164, "ot_trace_id": "09f23ffa86e23002", "ot_span_id": "09f23ffa86e23003", "ot_start": "2022-09-07T17:32:19.355170Z", "ot_stop": "2022-09-07T17:32:19.355481Z", "ot_duration": "0.311ms", "ot_child_of": "09f23ffa86e23001"}
2022-09-07T17:32:19.379656Z     info    bolt.(*KVStore).View    {"log_id": "0cnFyGv0000", "filename": "github.com/influxdata/influxdb/v2/bolt/kv.go", "line": 164, "ot_trace_id": "09f23ffa86e23002", "ot_span_id": "09f23ffa87223000", "ot_start": "2022-09-07T17:32:19.356352Z", "ot_stop": "2022-09-07T17:32:19.379652Z", "ot_duration": "23.308ms", "ot_child_of": "09f23ffa86e23001"}
2022-09-07T17:32:19.379900Z     info    bolt.(*KVStore).View    {"log_id": "0cnFyGv0000", "filename": "github.com/influxdata/influxdb/v2/bolt/kv.go", "line": 164, "ot_trace_id": "09f23ffa86e23002", "ot_span_id": "09f23ffa8ce23000", "ot_start": "2022-09-07T17:32:19.379882Z", "ot_stop": "2022-09-07T17:32:19.379898Z", "ot_duration": "0.016ms", "ot_child_of": "09f23ffa86e23001"}
2022-09-07T17:32:19.380007Z     error   api error encountered   {"log_id": "0cnFyGv0000", "handler": "user", "error": "invalid user ID provided in route"}
2022-09-07T17:32:19.380034Z     error   api error encountered   {"log_id": "0cnFyGv0000", "handler": "user", "error": "id must have a length of 16 bytes"}
2022-09-07T17:32:19.380149Z     info    http: superfluous response.WriteHeader call from github.com/influxdata/influxdb/v2/kit/transport/http.(*StatusResponseWriter).WriteHeader (status_response_writer.go:26)        {"log_id": "0cnFyGv0000", "service": "http"}
2022-09-07T17:32:19.380163Z     info    request {"log_id": "0cnFyGv0000", "path": "/api/v2/me/password", "filename": "net/http/request.go", "line": 356, "error.object": "no span ID found in carrier", "user_agent": "curl", "Accept": "*/*", "Content-Type": "application/json", "Content-Length": "31", "Cookie": "influxdb-oss-session=GU6DzwdrK7GFo8D0CAxECQlSxkTViMlO1hscEn67S7vZ1dGcHkjv84vrllzLlq7YGHtzAWRSqoGhfwe9l0pwsg==", "appMetrics": false, "queryTracing": false, "injectLatestSuccessTime": false, "enforceOrgDashboardLimits": false, "narrowTransformationLimit": false, "memoryOptimizedSchemaMutation": false, "newAutoRefresh": true, "groupTransformationGroup": false, "optimizeAggregateWindow": false, "newLabels": false, "narrowTransformationDifference": false, "narrowTransformationFill": false, "timeFilterFlags": false, "optimizeUnionTransformation": false, "optimizeStateTracking": false, "vectorizeAddition": false, "cursorAtEOF": false, "refreshSingleCell": true, "vectorizedMap": false, "unusedSymbolWarnings": false, "memoryOptimizedFill": false, "aggregateTransformationTransport": false, "vectorizeOperators": false, "groupWindowAggregateTranspose": false, "queryConcurrencyLimit": 0, "vectorizeLogicalOperators": false, "labelPolymorphism": false, "optimizeSetTransformation": false, "ot_trace_id": "09f23ffa86e23002", "ot_span_id": "09f23ffa86e23001", "ot_start": "2022-09-07T17:32:19.355067Z", "ot_stop": "2022-09-07T17:32:19.380160Z", "ot_duration": "25.102ms", "method": "PUT", "route": "", "handler": "platform", "user_id": "0772396d1f411000"}
2022-09-07T17:32:21.685188Z     info    bolt.(*KVStore).View    {"log_id": "0cnFyGv0000", "filename": "github.com/influxdata/influxdb/v2/bolt/kv.go", "line": 164, "ot_trace_id": "09f23ffccd623001", "ot_span_id": "09f23ffccd623000", "ot_start": "2022-09-07T17:32:21.685014Z", "ot_stop": "2022-09-07T17:32:21.685129Z", "ot_duration": "0.115ms"}
2022-09-07T17:32:31.681985Z     info    bolt.(*KVStore).View    {"log_id": "0cnFyGv0000", "filename": "github.com/influxdata/influxdb/v2/bolt/kv.go", "line": 164, "ot_trace_id": "09f2400690623001", "ot_span_id": "09f2400690623000", "ot_start": "2022-09-07T17:32:31.681819Z", "ot_stop": "2022-09-07T17:32:31.681978Z", "ot_duration": "0.159ms"}
2022-09-07T17:32:41.680872Z     info    bolt.(*KVStore).View    {"log_id": "0cnFyGv0000", "filename": "github.com/influxdata/influxdb/v2/bolt/kv.go", "line": 164, "ot_trace_id": "09f2401054223001", "ot_span_id": "09f2401054223000", "ot_start": "2022-09-07T17:32:41.680806Z", "ot_stop": "2022-09-07T17:32:41.680858Z", "ot_duration": "0.052ms"}
2022-09-07T17:32:51.681200Z     info    bolt.(*KVStore).View    {"log_id": "0cnFyGv0000", "filename": "github.com/influxdata/influxdb/v2/bolt/kv.go", "line": 164, "ot_trace_id": "09f2401a18623001", "ot_span_id": "09f2401a18623000", "ot_start": "2022-09-07T17:32:51.681132Z", "ot_stop": "2022-09-07T17:32:51.681185Z", "ot_duration": "0.053ms"}
2022-09-07T17:33:00.039637Z     info    bolt.(*KVStore).Update  {"log_id": "0cnFyGv0000", "filename": "github.com/influxdata/influxdb/v2/bolt/kv.go", "line": 177, "ot_trace_id": "09f2402238223001", "ot_span_id": "09f2402238223000", "ot_start": "2022-09-07T17:33:00.000048Z", "ot_stop": "2022-09-07T17:33:00.039628Z", "ot_duration": "39.581ms"}
2022-09-07T17:33:00.039810Z     info    bolt.(*KVStore).View    {"log_id": "0cnFyGv0000", "filename": "github.com/influxdata/influxdb/v2/bolt/kv.go", "line": 164, "ot_trace_id": "09f2402241e23001", "ot_span_id": "09f2402241e23002", "ot_start": "2022-09-07T17:33:00.039712Z", "ot_stop": "2022-09-07T17:33:00.039799Z", "ot_duration": "0.087ms", "ot_child_of": "09f2402241e23000"}
2022-09-07T17:33:00.040379Z     info    bolt.(*KVStore).View    {"log_id": "0cnFyGv0000", "filename": "github.com/influxdata/influxdb/v2/bolt/kv.go", "line": 164, "ot_trace_id": "09f2402241e23001", "ot_span_id": "09f2402241e23003", "ot_start": "2022-09-07T17:33:00.039833Z", "ot_stop": "2022-09-07T17:33:00.040369Z", "ot_duration": "0.536ms", "ot_child_of": "09f2402241e23000"}
(edited)
```

### PR DESCRIPTION
**Pull request by jeffreyssmith2nd**
- Closes #23716

### Description
There were a whole host of discrepancies between how `/me` is supposed to work and how it was actually working. This PR should bring it (and `/users`) back up to spec with the docs/cloud 2.

### Note for reviewers:
Check the semantic commit type:
 - Feat: a feature with user-visible changes
 - Fix: a bug fix that we might tell a user “upgrade to get this fix for your issue”
 - Chore: version bumps, internal doc (e.g. README) changes, code comment updates, code formatting fixes… must not be user facing (except dependency version changes)
 - Build: build script changes, CI config changes, build tool updates
 - Refactor: non-user-visible refactoring
 - Check the PR title: we should be able to put this as a one-liner in the release notes

### CODE REVIEW

File: tenant/http_server_user.go [hunk 1]
```diff
@@ -67,20 +66,8 @@ func NewHTTPUserHandler(log *zap.Logger, userService influxdb.UserService, passw
...
-	return h.prefix
-}
-func (h *UserHandler) MeResourceHandler() *resourceHandler {
-	return &resourceHandler{prefix: prefixMe, UserHandler: h}
-}
-
-func (h *UserHandler) UserResourceHandler() *resourceHandler {
-	return &resourceHandler{prefix: prefixUsers, UserHandler: h}
+func (h *UserHandler) Prefix() string {
+	return prefixUsers
```
Comment (davidby-influx) [inline tenant/http_server_user.go:70]: You could move the `const` declaration in here to reduce its visibility.

File: tenant/http_server_user.go [hunk 2]
```diff
@@ -495,3 +446,87 @@ func decodePatchUserRequest(ctx context.Context, r *http.Request) (*patchUserReq
...
+	ctx := r.Context()
+	userID, err := h.getUserID(ctx)
+	if err != nil {
+		h.api.Err(w, r, err)
+	}
+	user, err := h.userSvc.FindUserByID(ctx, *userID)
+	if err != nil {
+		h.api.Err(w, r, err)
+	}
+	h.api.Respond(w, r, http.StatusOK, newUserResponse(user))
```
Comment (davidby-influx) [inline tenant/http_server_user.go:508]: On error, does this respond with the err, then the StatusOk?  Should there be a return after the `h.Api.Err`  calls?

**Review by davidby-influx — CHANGES_REQUESTED** [review]
Slightly confused about error handling, probably my own ignorance.

File: tenant/http_server_user.go (hunk 2 above)
Comment (jeffreyssmith2nd) [reply tenant/http_server_user.go:508]: You are correct, it is missing the return. Must have missed that when I was splicing code around.

Comment (davidby-influx) [reply tenant/http_server_user.go:508]: There are quite a few of those in the code, not just in this method.

File: tenant/http_server_user.go (hunk 1 above)
Comment (davidby-influx) [reply tenant/http_server_user.go:70]: Should the other uses of the consts be these method calls, or is the containing object not available there?

**Review by davidby-influx — APPROVED** [review]
LGTM

**Pull request merged**